src/app/pages/LendingPage/components/LendingDialog/index.tsx

```diff
@@ -94,15 +101,33 @@ export const LendingDialog: React.FC<ILendingDialogProps> = ({
     if (isTotalClicked) {
       withdrawAmount = depositedAssetBalance;
     }
+    if (bignumber(marketLiquidity).lessThanOrEqualTo(withdrawAmount)) {
+      withdrawAmount = marketLiquidity;
+    }
+
     return bignumber(withdrawAmount)
       .mul(bignumber(depositedBalance).div(depositedAssetBalance))
       .toFixed(0);
-  }, [depositedAssetBalance, depositedBalance, isTotalClicked, weiAmount]);
+  }, [
+    depositedAssetBalance,
+    depositedBalance,
+    isTotalClicked,
+    weiAmount,
+    marketLiquidity,
+  ]);
 
   const [withdrawAmount, setWithdrawAmount] = useState(
     calculateWithdrawAmount(),
   );
 
+  const maxWithdrawAmount = useMemo(() => {
+    if (bignumber(depositedAssetBalance).greaterThan(marketLiquidity)) {
```

Comment:
Same logic should be applied to Withdraw button disabled state on line 92 in `validate` variable (currently the user is prevented from pushing through a tx which will fail, but the button is still enabled which is confusing)